modules/jdbc/src/main/java/org/apache/ignite/internal/jdbc/JdbcResultSet.java

```diff
@@ -99,23 +98,22 @@ protected DecimalFormat initialValue() {
         }
     };
 
-    /** Statement. */
     private final JdbcStatement stmt;
-
-    /** Cursor ID. */
-    private final Long cursorId;
+    private final @Nullable Long cursorId;
+    private final boolean hasResultSet;
+    private final boolean hasNextResult;
 
     /** Column order map. */
-    private Map<String, Integer> colOrder;
+    private @Nullable Map<String, Integer> colOrder;
 
     /** Rows. */
-    private List<BinaryTupleReader> rows;
+    private @Nullable List<BinaryTupleReader> rows;
 
     /** Rows iterator. */
-    private Iterator<List<Object>> rowsIter;
+    private @Nullable Iterator<List<Object>> rowsIter;
 
     /** Current row. */
-    private List<Object> curRow;
+    private @Nullable List<Object> curRow;
```

Comment:
seems we need to add nullability check for such a case ?

Reply:
done